appserver/web/web-glue/src/main/java/com/sun/enterprise/web/VirtualServer.java

```diff
@@ -920,109 +925,55 @@ synchronized void setLogFile(String logFile, String logLevel, String logServiceF
-                    @Override
-                    public ResourceBundle getResourceBundle() {
-                        return rb;
-                    }
-
-                    @Override
-                    public synchronized void addHandler(Handler handler) {
-                        super.addHandler(handler);
-                        if (handler instanceof FileLoggerHandler) {
-                            ((FileLoggerHandler) handler).associate();
-                        }
-                    }
-
-                    @Override
-                    public synchronized void removeHandler(Handler handler) {
-                        if (!(handler instanceof FileLoggerHandler)) {
-                            super.removeHandler(handler);
-                        } else {
-                            boolean hasHandler = false;
-                            Handler[] hs = getHandlers();
-                            if (hs != null) {
-                                for (Handler h : hs) {
-                                    if (h == handler) {
-                                        hasHandler = true;
-                                        break;
-                                    }
-                                }
-                            }
-                            if (hasHandler) {
-                                super.removeHandler(handler);
-                                ((FileLoggerHandler) handler).disassociate();
-                            }
-                        }
-                    }
-                };
-
-                synchronized (Logger.class) {
-                    LogManager.getLogManager().addLogger(newLogger);
-                }
-            }
+            setLogger(newLogger);
+            return;
+        }
 
-            // remove old handlers if necessary
-            Handler[] handlers = newLogger.getHandlers();
-            if (handlers != null) {
-                for (Handler h : handlers) {
-                    newLogger.removeHandler(h);
-                }
+        // remove old handlers if necessary
+        // FIXME: does not respect the configuration - why?
+        final Handler[] originalLoggerHandlers = newLogger.getHandlers();
+        if (originalLoggerHandlers != null) {
+            for (Handler h : originalLoggerHandlers) {
+                newLogger.removeHandler(h);
             }
+        }
 
-            // add handlers from root that is not GFFileHandler
-            Logger rootLogger = Logger.global.getParent();
-            if (rootLogger != null) {
-                Handler[] rootHandlers = rootLogger.getHandlers();
-                if (rootHandlers != null) {
-                    for (Handler h : rootHandlers) {
-                        if (!(h instanceof GFFileHandler)) {
-                            newLogger.addHandler(h);
-                        }
+        // FIXME: takes handlers only from root logger! What about intermediate?
```

Comment:
Another one

Reply:
Both FIXMEs relate to the old code, I was asked to stop at some point, so I created standalone issues for these things. If I would leave just a comment here, nobody would pay attention on that, so you proved now that FIXMEs attract interest (and yes, they don't resolve issues, that's the point why they should not be in the code - no FIXMEs and no bugs, but this is a bug, luckily nobody uses it).

- any FIXME or TODO in the code means "this is wrong" and must be resolved asap
- see FISH-1298 for more, maybe next time I will create issue first, so I could mention it in the comment.
- the point is that it doesn't work at all for a long time. The log file is created, but is always empty.
- yet one thing - the logging of the server is nonstandard and limited. We should slowly push it to be able to use logging in some standard way. It is not so far, but the technical debt and dead features should be resolved first. After this PR logging should be at least helpful.